gcc/testsuite/rust/execute/torture/issue-2583.rs

```diff
@@ -0,0 +1,13 @@
+#[lang = "sized"]
+pub trait Sized {}
+
+mod intrinsics {
+    extern "rust-intrinsic" {
+        pub fn uninit<T>() -> T;
+    }
+}
+
+pub fn main() -> usize {
+    let val: usize = unsafe { intrinsics::uninit() };
+    val - 1
```

Comment:
should we guarantee that `val` will be initialized to `1` here? is that something that Rust guarantees?

Reply:
Good point

Reply:
because actually all i care about is that this test case doesn't segv at runtime